### ISSUE TITLE
Any role that can create users can create Super Admin user. 

### ISSUE THREAD

**Issue by branislav-brincko** [strapi/strapi#16297]
### Required System information

- Node.js version: 16.17.1
- NPM version: 9.1.3
- Strapi version: 4.9.0
- Database: Postgresql
- Operating system: macOS 12.5.1
- Is your project Javascript or Typescript: Typescript

### Describe the bug

Strapi made "Custom Roles & Permissions" available for Free in Strapi v4.8. I am dealing with the exactly same issue as nicely described here on Strapi forum:
https://forum.strapi.io/t/strapi-role-hierarchy-restrict-the-ability-to-create-new-users-with-the-superadmin-role-for-users-of-a-custom-role/4691

TLDR: user with custom role, that can create users, can create ANY kind of user, including Super Admin, which is security problem.

**Comment by utkarshk384** [comment]
Is anyone working on this issue? If not I'll work on it with some guidance.
